File: src/RqCalc.WebApp/_Scripts/Models/GuildTalentModel.js
function GuildTalentModel(context, branchModel, talent) {

    var thisModel = this;

    thisModel.active = ko.observable();

    thisModel.image = ko.observable();

    thisModel.talent = talent;

    thisModel.mainDescription = ko.observable();

    thisModel.points = ko.observable(0);

    thisModel.points.subscribe(pointsChanged);

    thisModel.branch = branchModel;
    
    updateDescription();

    function pointsChanged(points) {
        thisModel.active(points > 0);

        updateDescription();
    }

    function updateDescription() {

        var description = MainHelper.TextTemplate.getGuildTalentDescription(talent, Math.max(1, thisModel.points()));

        var newDescriptionModel = new TalentDescriptionModel(context, [{ key: TextTemplateVariableType.const, value: 1 }], description);

        thisModel.mainDescription(newDescriptionModel);    
    }
    
    thisModel.active.subscribe(function(active) {

        thisModel.image(active ? talent.image : talent.grayImage);
    });

    thisModel.active(false);
    
    thisModel.increase = function () {

        branchModel.windowModel.change(this, true);
    }

    thisModel.decrease = function () {

        branchModel.windowModel.change(this, false);
    }
}

File: src/RqCalc.Wpf/Models/GuildTalentModel.cs
using Framework.Reactive;
using Framework.Reactive.ObservableRecurse;
using RqCalc.Domain;
using RqCalc.Domain.GuildTalent;
using RqCalc.Model._Extensions;
using RqCalc.Wpf.Models._Base;

namespace RqCalc.Wpf.Models;

public class GuildTalentModel : ActiveImageChangeModel<IGuildTalent>
{
    public readonly GuildTalentBranchModel Branch;


    public GuildTalentModel(GuildTalentBranchModel branch, IGuildTalent talent)

    {
        this.Branch = branch ?? throw new ArgumentNullException(nameof(branch));
        this.SelectedObject = talent ?? throw new ArgumentNullException(nameof(talent));
            
        this.SubscribeExplicit(rootRule => rootRule.Subscribe(model => model.Points, this.PointsChanged));
        this.UpdateDescription();
    }


    public TalentDescriptionModel MainDescription
    {
        get => this.GetValue(v => v.MainDescription);
        private set => this.SetValue(v => v.MainDescription, value);
    }

    public bool HasPassive { get; } = false;

    public bool HasEquipmentCondition { get; } = false;

    public int Points
    {
        get => this.GetValue(v => v.Points);
        set => this.SetValue(v => v.Points, value);
    }

    private void PointsChanged()
    {
        this.Active = this.Points > 0;

        this.UpdateDescription();
    }

    private void UpdateDescription() => this.MainDescription = new TalentDescriptionModel(new Dictionary<TextTemplateVariableType, decimal> { { TextTemplateVariableType.Const, 1 } }, this.SelectedObject.GetDescription(Math.Max(1, this.Points)));

    public void Change(bool increase) => this.Branch.WindowModel.Change(this, increase);
}